EIPS/eip-4675.md

```diff
@@ -0,0 +1,212 @@
+    */
+    function totalSupply(uint256 _id) external view returns (uint256);
+
+    /**
+        @notice Get the balance of an account's tokens.
+        @param _owner  The address of the token holder
+        @param _id     ID of the token
+        @return        The _owner's balance of the token type requested
+    */
+    function balanceOf(address _owner, uint256 _id) external view returns (uint256);
+
+    /**
+        @notice Get the amount which `_spender` is still allowed to withdraw from `_owner`
+        @param _owner   The address of the token holder
+        @param _spender The address approved to withdraw token on behalf of `_owner`
+        @param _id      ID of the token
+        @return         The amount which `_spender` is still allowed to withdraw from `_owner`
+    */
+    function allowance(address _owner, address _spender, uint256 _id) external view returns (uint256);
+
+    /**
+        @notice Get the bool value which represents whether the NFT is already registered and fractionalized by this contract.
+        @param _parentNFTContractAddress    NFT contract address
+        @param _parentNFTTokenId            NFT tokenID
+        @return                             The bool value representing the whether the NFT is already registered.
+    */
+    function isRegistered(address _parentNFTContractAddress, uint256 _parentNFTTokenId) external view returns (bool);
+}
+
+interface ERC165 {
+    /**
+        @notice Query if a contract implements an interface
+        @param interfaceID The interface identifier, as specified in ERC-165
+        @dev Interface identification is specified in ERC-165. This function
+        uses less than 30,000 gas.
+        @return `true` if the contract implements `interfaceID` and
+        `interfaceID` is not 0xffffffff, `false` otherwise
+    */
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+```
+
+To receive Non-Fungible Token on `safe Transfer` the contract should include `onERC721Received()`.
+Including `onERC721Received()` is needed to be compatible with Safe Transfer Rules.
+```solidity
+/**
+    @notice Handle the receipt of an NFT
+    @param _operator The address which called `safeTransferFrom` function
+    @param _from The address which previously owned the token
+    @param _tokenId The NFT identifier which is being transferred
+    @param _data Additional data with no specified format
+    @return `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
+*/
+function onERC721Received(address _operator, address _from, uint256 _tokenId, bytes calldata _data) external pure returns (bytes4);
+```
+
+## Rationale
+
+**Metadata**
+
+The `symbol()` & `name()` function was not included since the majority of users can just fetch it from the originating NFT contract. Also, copying the name & symbol every time when token gets added might place a lot of redundant bytecode on the Ethereum blockchain. 
```

Comment:
```suggestion
The `symbol()` & `name()` functions were not included since the majority of users can just fetch it from the originating NFT contract. Also, copying the name & symbol every time when token gets added might place a lot of redundant bytecode on the Ethereum blockchain. 
```